test/Dialect/Arc/make-tables.mlir

```diff
@@ -0,0 +1,66 @@
+// RUN: circt-opt %s --arc-make-tables | FileCheck %s
+
+arc.define @test_arc(%arg0: i4) -> i4 {
+  %c0_i3 = hw.constant 0 : i3
+  %c0_i2 = hw.constant 0 : i2
+  %false = hw.constant false
+  %0 = comb.concat %c0_i3, %arg0 : i3, i4
+  %1 = comb.concat %c0_i2, %arg0 : i2, i4
+  %2 = comb.concat %false, %arg0 : i1, i4
+  %3 = comb.concat %arg0, %false : i4, i1
+  %4 = comb.and bin %2, %3 : i5
+  %5 = comb.concat %false, %4 : i1, i5
+  %6 = comb.add bin %1, %5 : i6
+  %7 = comb.and bin %1, %6 : i6
+  %8 = comb.concat %false, %7 : i1, i6
+  %9 = comb.add bin %0, %8 : i7
+  %10 = comb.extract %9 from 0 : (i7) -> i4
+  %11 = comb.and %arg0, %10 : i4
+  %12 = comb.add %arg0, %11 : i4
+  %13 = comb.and %arg0, %12 : i4
+  %14 = comb.add %arg0, %13 : i4
+  %15 = comb.and %arg0, %14 : i4
+  %16 = comb.add %arg0, %15 : i4
+  %17 = comb.and %arg0, %16 : i4
+  %18 = comb.add %arg0, %17 : i4
+  %19 = comb.and %arg0, %18 : i4
+  %20 = comb.add %arg0, %19 : i4
+  %21 = comb.and %arg0, %20 : i4
+  arc.output %21 : i4
+}
+hw.module @test(%clock: i1, %x: i4) -> (out: i4) {
+  %c0_i3 = hw.constant 0 : i3
+  %false = hw.constant false
+  %c0_i2 = hw.constant 0 : i2
+  %0 = arc.state @test_arc(%x) lat 0 : (i4) -> i4
+  hw.output %0 : i4
+}
+
+// CHECK-LABEL: arc.define @test_arc(%arg0: i4) -> i4 {
+// CHECK-NEXT:   %c7_i4 = hw.constant 7 : i4
+// CHECK-NEXT:   %c-2_i4 = hw.constant -2 : i4
+// CHECK-NEXT:   %c-3_i4 = hw.constant -3 : i4
+// CHECK-NEXT:   %c-4_i4 = hw.constant -4 : i4
+// CHECK-NEXT:   %c-5_i4 = hw.constant -5 : i4
+// CHECK-NEXT:   %c-6_i4 = hw.constant -6 : i4
+// CHECK-NEXT:   %c-7_i4 = hw.constant -7 : i4
+// CHECK-NEXT:   %c-8_i4 = hw.constant -8 : i4
+// CHECK-NEXT:   %c7_i4_0 = hw.constant 7 : i4
+// CHECK-NEXT:   %c6_i4 = hw.constant 6 : i4
+// CHECK-NEXT:   %c5_i4 = hw.constant 5 : i4
+// CHECK-NEXT:   %c4_i4 = hw.constant 4 : i4
+// CHECK-NEXT:   %c3_i4 = hw.constant 3 : i4
+// CHECK-NEXT:   %c2_i4 = hw.constant 2 : i4
+// CHECK-NEXT:   %c1_i4 = hw.constant 1 : i4
+// CHECK-NEXT:   %c0_i4 = hw.constant 0 : i4
+// CHECK-NEXT:   %0 = hw.array_create %c7_i4, %c-2_i4, %c-3_i4, %c-4_i4, %c-5_i4, %c-6_i4, %c-7_i4, %c-8_i4, %c7_i4_0, %c6_i4, %c5_i4, %c4_i4, %c3_i4, %c2_i4, %c1_i4, %c0_i4 : i4
```

Comment:
Nit: Now we have a `aggregate_constant` op that we could insert directly.

Reply:
The pass emits that now :tada: